dashboard/app/controllers/courses_controller.rb

```diff
@@ -20,4 +29,21 @@ def show
 
     render 'show', locals: {course: course}
   end
+
+  def new
+    render 'new'
+  end
+
+  def create
+    course = Course.new(name: params.require(:course).require(:name))
+    if course.save
+      redirect_to action: :edit, course_name: course.name
+    else
+      render 'new', locals: {error_messages: course.errors.full_messages}
```

Comment:
I've been trying to avoid using instance variables in my views, as I find them much more difficult to follow than explicitly passing things in via locals. This might be a case where there's a good argument for doing it the way you say.